### PR TITLE
doc: use :zephyr_file: where appropriate

### PR DESCRIPTION
**Pull request by dbkinder**
A new role :zephyr_file: is available that renders to a link to the file
or folder in GitHub.  Find appropriate references using :file: and convert
to :zephyr_file: to take advantage of its linking capability.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

**Comment by dbkinder** [comment]
P.S.  I've also verified that all the files referenced by :zephyr_file: roles actually exist

**Review by nashif — APPROVED** [review]

File: doc/guides/documentation/index.rst [hunk 1]
```diff
@@ -246,6 +246,12 @@ For example, there are roles for marking :file:`filenames`
 (``:command:`make```).  You can also use the \`\`inline code\`\`
 markup (double backticks) to indicate a ``filename``.
 
+For references to files that are in the Zephyr GitHub tree, a special
```
Comment (aescolar) [inline doc/guides/documentation/index.rst]: or folders?

**Review by aescolar — COMMENTED** [review]
Neat. Just a very minor comment

**Review by jukkar — APPROVED** [review]
Thanks David, I had plans to do something like this but now I do not have to.

**Comment by pfalcon** [comment]
This is absolutely cool, was on my "can we do that??" list for a while. Is there a chance to still render the contents of the tag using a monospace font (i.e. like :file: has it)?

**Pull request merged**